### PR TITLE
[AIRFLOW-XXX] Group references in one section

### PR DESCRIPTION
**Pull request by mik-laj**
Preview is available: http://frantic-station.surge.sh/ (10 Aug 2019, 15:26 CEST)

---

Make sure you have checked _all_ steps below.

### Jira

- [ ] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-XXX
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [ ] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### Code Quality

- [ ] Passes `flake8`

### CODE REVIEW

File: docs/usage-rest-api.rst [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    under the License.
+
+Using the REST API
+==================
+
+This document is meant to give a overview of all common tasks while using an REST API.
+
+.. note::
+
+   This function is not suitable for sending spam e-mails.
```
Comment (ashb) [inline docs/usage-rest-api.rst]: ?

File: docs/usage-rest-api.rst [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+.. code-block:: ini
+
+    [api]
+    auth_backend = airflow.api.auth.backend.deny_all
+
+Two "real" methods for authentication are currently supported for the API.
+
+To enabled Password authentication, set the following in the configuration:
+
+.. code-block:: bash
```
Comment (ashb) [inline docs/usage-rest-api.rst]: This was bash before, but that looks like a mistake and we should fix it to `ini`

File: docs/usage-rest-api.rst (hunk 1 above)
Comment (mik-laj) [reply docs/usage-rest-api.rst]: Och. Mistake

File: docs/usage-rest-api.rst (hunk 2 above)
Comment (mik-laj) [reply docs/usage-rest-api.rst]: Fixed.

File: docs/usage-rest-api.rst (hunk 1 above)
Comment (mik-laj) [reply docs/usage-rest-api.rst]: Fixed.

File: docs/usage-rest-api.rst [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+Using the REST API
+==================
+
+This document is meant to give a overview of all common tasks while using an REST API.
```
Comment (kaxil) [inline docs/usage-rest-api.rst]:
```suggestion
This document is meant to give an overview of all common tasks while using the REST API.
```

File: docs/usage-rest-api.rst [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+.. note::
+
+    API endpoints and samples are described in :doc:`rest-api-ref`.
+
+CLI
+-----
```
Comment (kaxil) [inline docs/usage-rest-api.rst]:
```suggestion
---
```

File: docs/usage-rest-api.rst [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    api_client = airflow.api.client.json_client
+    endpoint_url = http://<WEBSERVER>:<PORT>
+
+
+Authentication
+--------------
+
+Authentication for the API is handled separately to the Web Authentication. The default is to not
+require any authentication on the API -- i.e. wide open by default. This is not recommended if your
+Airflow webserver is publicly accessible, and you should probably use the deny all backend:
```
Comment (kaxil) [inline docs/usage-rest-api.rst]:
```suggestion
Airflow webserver is publicly accessible, and you should probably use the ``deny_all`` backend:
```

**Review by kaxil — APPROVED** [review]
nit - Minor suggestions. Feel free to merge after those changes are implemented.

**Pull request merged**